0.1-test-notebook.ipynb

```diff
@@ -0,0 +1,200 @@
+    "\n",
+    "Below is a code block. You can double click to edit it. You can also run the code step-by-step (one cell a time) by pressing shift + enter when you have the cell selected."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "21"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "14+7"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Configure your source directory\n",
+    "\n",
+    "In the `setup.py` file, enter the source directory for your local Optech taproot (v0.1 tag) bitcoin repo.  \n",
+    "This will be used to access the TestFramework Python code and the taproot-compatible bitcoind binary."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Source directory configured as /Users/schmidty/Dropbox/Repos/bitcoin-optech-taproot\n"
+     ]
+    }
+   ],
+   "source": [
+    "import setup"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Import test framework\n",
+    "\n",
+    "We provide a test wrapper class which allows the user to leverage the test framework utilites in external Python projects, such as this notebook. The test wrapper provides access to bitcoin rpc instances directly from the Jupyter notebook cell or any other external python project.\n",
+    "\n",
+    "When `setup` is called, the test wrapper will automatically startup a standard of 3 bitcoind daemon subprocesses, which can be accessed through the included rpc interface. We will use this frequently in this workshop to broadcast schnorr & taproot transactions.\n",
+    "\n",
+    "![test_wrapper_object_lifetime](files/0.1-test-notebook/test-wrapper-object-lifetime.jpg)\n",
+    "\n",
+    "The import below requires the Optech taproot branch to be checked out. If it fails, then you may have entered the source directory above incorrectly or checked out the wrong branch."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "2019-08-29T19:30:18.841000Z TestFramework./var/folders/fq/cbxk_q2n2sb1hw8zxl987v7h0000gn/T/bitcoin_func_test_yedz4z_d (INFO): Initializing test directory /var/folders/fq/cbxk_q2n2sb1hw8zxl987v7h0000gn/T/bitcoin_func_test_yedz4z_d\n"
+     ]
+    }
+   ],
+   "source": [
+    "from test_framework.test_wrapper import TestWrapper\n",
```

Comment:
If using util.py, this import can be removed.

Reply:
Done. I've rebased this PR on the TestWrapper changes.